### ISSUE TITLE
HibernateMetrics fails if Hibernate is not on the classpath

### ISSUE THREAD

**Issue by snicoll** [micrometer-metrics/micrometer#1037]
Already reported in #518 but I didn't properly follow-up.

The intention that is discussed there ([this code](https://github.com/micrometer-metrics/micrometer/blob/6cdc476985b2f6e57d0c2487dd0745e7d02aabd6/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jpa/HibernateMetrics.java#L170-L185)) does not work if Hibernate is not on the classpath.

```
Caused by: java.lang.NoClassDefFoundError: org/hibernate/SessionFactory
    at io.micrometer.core.instrument.binder.jpa.HibernateMetrics.getStatistics(HibernateMetrics.java:180)
    at io.micrometer.core.instrument.binder.jpa.HibernateMetrics.hasStatisticsEnabled(HibernateMetrics.java:166)
    at io.micrometer.core.instrument.binder.jpa.HibernateMetrics.<init>(HibernateMetrics.java:57)
    at org.springframework.boot.actuate.autoconfigure.metrics.orm.jpa.HibernateMetricsAutoConfiguration.bindEntityManagerFactoryToRegistry(HibernateMetricsAutoConfiguration.java:71)
```

There was a faulty condition in Spring Boot that got fixed in https://github.com/spring-projects/spring-boot/issues/15317 but something should probably change here as well. 

To make it more obvious that a `SessionFactory` is required, perhaps the constructor should require that rather than an `EntityManagerFactory` that doesn't enforce anything related to Hibernate. Alternatively, `HibernateMetrics` could gracefully handle the fact that Hibernate is not on the classpath.

**Comment by izeye** [comment]
Looking at the code, it seems to assume that Hibernate is on the classpath as it has [a final member field typed `Statistics`](https://github.com/micrometer-metrics/micrometer/blob/master/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jpa/HibernateMetrics.java#L45) which comes from Hibernate. As @snicoll suggested, exposing `SessionFactory` might be an option to be clear or updating Javadoc might be another option.

For https://github.com/spring-projects/spring-boot/issues/15317, I created https://github.com/micrometer-metrics/micrometer/pull/1039 to fix in the Spring Boot legacy support.

**Comment by izeye** [comment]
Reopening as @snicoll 's original request doesn't seem to be resolved or answered yet. I guess the referenced fix for Spring Boot legacy support confuse @jkschneider .

**Comment by izeye** [comment]
Removing the milestone as this doesn't need to be resolved or answered within the milestone.

**Comment by izeye** [comment]
This seems resolved via #1053.
